src/whitenoise/accounts/groups.rs

```diff
@@ -78,10 +77,16 @@ impl Whitenoise {
             .map(|g| hex::encode(g.nostr_group_id))
             .collect::<Vec<_>>();
 
+        if welcome_rumors.len() != contacts.len() {
+            return Err(WhitenoiseError::Other(anyhow::Error::msg(
+                "Welcome rumours are missing for some of the members",
+            )));
+        }
```

Comment:
Should this check be before the `let group_ids = nostr_mls` line? Just so we 'bail out' early. I'd move the group_ids thing towards the end of the function closer to where it's actually used.